tests/basictests.cpp

```diff
@@ -24,479 +111,549 @@ inline uint64_t f64_ulp_dist(double a, double b) {
       }
       count++;
     }
-    if(count != n_records) {
+    if (count != n_records) {
       printf("Something is wrong in stream_test at window size = %zu.\n", i);
       return false;
     }
   }
-  printf("ok\n");
   return true;
 }
 
+// -------------------------------------------------------------------------------------------------
 // returns true if successful
 bool stream_test() {
-  printf("Running stream_test");
-  fflush(NULL);
+
   const size_t n_records = 10000;
-  std::string data;
-  char buf[1024];
+  std::string data{};
+  char buf[1024]{};
   for (size_t i = 0; i < n_records; ++i) {
-    auto n = sprintf(buf,
-                     "{\"id\": %zu, \"name\": \"name%zu\", \"gender\": \"%s\", "
-                     "\"ete\": {\"id\": %zu, \"name\": \"eventail%zu\"}}",
-                     i, i, (i % 2) ? "homme" : "femme", i % 10, i % 10);
+    auto n = sprintf(
+        buf,
+        R"({"id": %zu, "name": "name%zu", "gender": "%s","ete": {"id": %zu, "name": "eventail%zu"}})",
+        i, i, (i % 2) ? "homme" : "femme", i % 10, i % 10);
     data += std::string(buf, n);
   }
-  for(size_t i = 1000; i < 2000; i += (i>1050?10:1)) {
+  for (size_t i = 1000; i < 2000; i += (i > 1050 ? 10 : 1)) {
     printf(".");
-    fflush(NULL);
+
     simdjson::JsonStream js{data.c_str(), data.size(), i};
-    int parse_res = simdjson::SUCCESS_AND_HAS_MORE;
-    size_t count = 0;
-    simdjson::ParsedJson pj;
-    while (parse_res == simdjson::SUCCESS_AND_HAS_MORE) {
-      parse_res = js.json_parse(pj);
+    // int parse_res = simdjson::SUCCESS_AND_HAS_MORE;
+    size_t count{};
+    simdjson::ParsedJson pj{};
+    while (js.json_parse(pj) == simdjson::SUCCESS_AND_HAS_MORE) {
+      // parse_res = js.json_parse(pj);
       simdjson::ParsedJson::Iterator pjh(pj);
-      if(!pjh.is_object()) {
+      if (!pjh.is_object()) {
         printf("Root should be object\n");
         return false;
       }
-      if(!pjh.down()) {
+      if (!pjh.down()) {
         printf("Root should not be emtpy\n");
         return false;
       }
-      if(!pjh.is_string()) {
+      if (!pjh.is_string()) {
         printf("Object should start with string key\n");
         return false;
       }
-      if(strcmp(pjh.get_string(),"id")!=0) {
+      if (strcmp(pjh.get_string(), "id") != 0) {
         printf("There should a single key, id.\n");
         return false;
       }
       pjh.move_to_value();
-      if(!pjh.is_integer()) {
+      if (!pjh.is_integer()) {
         printf("Value of image should be integer\n");
         return false;
       }
       int64_t keyid = pjh.get_integer();
-      if(keyid != (int64_t)count) {
-        printf("key does not match %d, expected %d\n",(int) keyid, (int) count);
+      if (keyid != (int64_t)count) {
+        printf("key does not match %d, expected %d\n", (int)keyid, (int)count);
         return false;
       }
-      count++;
+
+      ++count;
     }
-    if(count != n_records) {
+    if (count != (n_records - 1)) {
```

Comment:
This has me concerned. The current (in master) check ensures that we have processed count == n_records. You will now test for the first  n_records - 1 values, omitting the last one. This is a change in the semantics of the test, and your new test is less stringent.

Reply:
When you code around midnight in the morning you are wondering who was coding :)